bilge-impl/src/try_from_bits.rs

```diff
@@ -24,12 +25,35 @@ fn parse(item: TokenStream) -> DeriveInput {
     shared::parse_derive(item)
 }
 
-fn analyze(derive_input: &DeriveInput) -> (&syn::Data, TokenStream, &Ident, BitSize) {
+fn analyze(derive_input: &DeriveInput) -> (&syn::Data, TokenStream, &Ident, BitSize, Option<Variant>) {
     shared::analyze_derive(derive_input, true)
 }
 
 fn analyze_enum(variants: Iter<Variant>, name: &Ident, internal_bitsize: BitSize) -> (Vec<TokenStream>, Vec<TokenStream>) {
-    shared::analyze_enum_derive(variants, name, internal_bitsize, true)
+    // in enums, internal_bitsize <= 64; u64::MAX + 1 = u128
+    if enum_fills_bitsize(internal_bitsize, variants.len()) {
+        emit_call_site_warning!("enum fills its bitsize"; help = "you can use `#[derive(FromBits)]` instead, rust will provide `TryFrom` for you (so you don't necessarily have to update call-sites)");
+    } 
+
+    let mut assigner = EnumVariantValueAssigner::new(internal_bitsize);
+    
+    variants.map(|variant| {
+        let variant_name = &variant.ident;
+        let variant_value = assigner.assign(variant);
```

Comment:
That's a really good idea, again :)
(removing low level enum handling from this high level code)